plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java

```diff
@@ -202,25 +208,50 @@ public static void waitForProjects(Runnable delayTactic, IProject... projects) {
         Collections.addAll(jobs, jobManager.find("org.eclipse.wst.server.core.family"));
         Collections.addAll(jobs, jobManager.find("org.eclipse.wst.server.ui.family"));
         Collections.addAll(jobs, jobManager.find(ValidationBuilder.FAMILY_VALIDATION_JOB));
+
+        // Remove SLEEPING jobs as they may not run for a long time.
```

Comment:
Tests are failing reliably, and I wonder if it has something to do with this. The Javadoc of `Job.schedule()` says

```
	 * Schedules this job to be run after a specified delay.  The job is put in the
	 * {@link #SLEEPING} state until the specified delay has elapsed, after which
	 * the job is added to a queue of {@link #WAITING} jobs. 
```
So, if an important job is scheduled with a delay, there is a chance we don't catch it. Not sure though.